bin/mangoplot.py

```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/env python
+
+import sys
+from pathlib import Path
+
+import csv
+
+import numpy as np
+
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Cursor
+
+from matplotlib.ticker import EngFormatter
+
+from typing import List, Union
+
+
+def identity(val):
+    return val
+
+
+def get_integer(val):
+    if is_integer(val):
+        return int(val)
+    else:
+        raise ValueError("Casting a non integer value: ", val)
+
+
+def is_integer(s):
+    try:
+        int(s)
+        return True
+    except ValueError:
+        return False
+
+
+def get_float(val):
+    if is_float(val):
+        return float(val)
+    else:
+        return float("nan")
+
+
+def is_float(s):
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
```

Comment:
Usually I don't write python, so take my comments with a grain of salt.

The above helpers should probably be `isinstance()` or wrappers around it, right?

Have you run this script through a linter - pylint, flake8, mypy to name a few - pylint throws flags 50+ issues.

Reply:
The problem these helpers are trying to solve is to know if a given `str` can be converted to `float` or `integer`. `isinstance` in this case cannot help as checks the type of the variable, in this case `s` is of type `str` :D

Yes, `flake8` gives me 5 line too long warnings that I decided to ignore, I will try pylint to see !

Reply:
Okay I fixed some of the things that `pylint` reported (the rest is like not following naming scheme for variables). We should be good now, please feel free to comment on anything you find weird/interesting.